validate-xml/validate-xml.cpp

```diff
@@ -0,0 +1,372 @@
+		char * e = XMLString::transcode(sig->getErrMsgs());
+		cout << e << endl;
+		XSEC_RELEASE_XMLCH(e);
+		return 0;
+	}
+	
+}
+	
+
+int main(int argc, const char** argv)
+{
+	if (argc < 2)
+	{
+		printUsage();
+		exit (1);
+	}
+	else
+	{
+		if(!IsFilePathValid(argv[1]))
+		{
+		cout<<"Invalid File";
+		exit(1);
+		}
+	}
+    try
+    {
+       XMLPlatformUtils::Initialize();
+	   XSECPlatformUtils::Initialise();
+    }
+    catch(const XMLException& e)
+    {
+        StrX tmp_e(e.getMessage());
+        cerr << "Initialization error: " << tmp_e.localForm() << endl;
+        return schemaError;
+    }
+
+    unique_ptr<XercesDOMParser> parser(new XercesDOMParser);
+    unique_ptr<DOMTreeErrorReporter> errReporter(new DOMTreeErrorReporter());
+    parser->setErrorHandler(errReporter.get());
+    parser->setDoNamespaces(true);
+    parser->useCachedGrammarInParse(true);
+    parser->setValidationScheme(XercesDOMParser::Val_Auto);
+    parser->setDoSchema(true);
+    parser->setValidationConstraintFatal(true);
+    parser->setCreateEntityReferenceNodes(true);
+
+	static Xerces::EntityFileResolver entityFileResolver;
+
+    if (entityFileResolver.IsEmpty())
+    {
+        entityFileResolver.AddFileContent("INTEROP_CPL_SCHEMA_NAME",
+                                          PROTO_ASDCP_CPL_20040511_XSD,
+                                          PROTO_ASDCP_CPL_20040511_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("XML_SCHEMA_NAME", XML_XSD, XML_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("SMPTE_CPL_SCHEMA_NAME", CPL_429_7_2006_XSD,
+                                          CPL_429_7_2006_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("XML_DSIG_SCHEMA_NAME", XMLDSIG_CORE_SCHEMA_XSD,
+                                          XMLDSIG_CORE_SCHEMA_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("SMPTE_CPL_STEREO_SCHEMA_NAME",
+                                          MAIN_STEREO_PICTURE_CPL_429_10_2008_XSD,
+                                          MAIN_STEREO_PICTURE_CPL_429_10_2008_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("INTEROP_CPL_STEREO_SCHEMA_NAME",
+                                          MAIN_STEREO_PICTURE_CPL_XSD,
+                                          MAIN_STEREO_PICTURE_CPL_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("INTEROP_CPL_CC_SCHEMA_NAME",
+                                          PROTO_ASDCP_CC_CPL_20070926_XSD,
+                                          PROTO_ASDCP_CC_CPL_20070926_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("SMPTE_TIMED_TEXT_SCHEMA_NAME", TT_429_12_2008_XSD,
+                                          TT_429_12_2008_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("CPL_DOLBY_ATMOS_AUX_DATA_SCHEMA_NAME",
+                                          DOLBY_AD_2012_XSD,
+                                          DOLBY_AD_2012_XSD_SIZE);
+    }
+
+    parser->setEntityResolver(&entityFileResolver);
```

Comment:
@RajaBoseLeoM Inorder to use setExternalSchemaLocation, I have added additional code. Initially I ignored the resolveEntity method's definition in the EntityFIleResolver.cpp file because it had a lot of references to Qubecpputils. But in order to add setExternalSchemaLocation, it seems that it is necessary and without it, it is throwing errors in parsing saying that "schemalocation does not contain namespace loacation pairs". So can you provide the qubecpputils definitions or advise me on how to proceed without it.